src/DataMiner/SLNetTypes/SubscriptionFilters/SubscriptionFilter.cs

```diff
@@ -0,0 +1,268 @@
+		/// </summary>
+		/// <param name="set1">First set of subscription filters.</param>
+		/// <param name="set2">Second set of subscription filters.</param>
+		/// <returns><c>true</c> if the specified sets are equal; otherwise, <c>false</c>.</returns>
+		public static bool EqualsSet(List<SubscriptionFilter> set1, List<SubscriptionFilter> set2)
+		{
+			return true;
+		}
+
+		/// <summary>
+		/// Gets or sets extra options, such as "skip initial events"
+		/// </summary>
+		/// <value>The extra options.</value>
+		public SubscriptionFilterOptions Options { get; set; }
+
+		/// <summary>
+		/// Determines whether the specified option is present.
+		/// </summary>
+		/// <param name="option">The option.</param>
+		/// <returns>
+		///   <c>true</c> if the specified option has option; otherwise, <c>false</c>.
+		/// </returns>
+		public bool HasOption(SubscriptionFilterOptions option)
+		{
+			return ((Options & option) == option);
+		}
+
+		/// <summary>
+		/// Returns a (deep) clone of this object 
+		/// </summary>
+		/// <returns>The clone.</returns>
+		public virtual Object Clone()
+		{
+			return new SubscriptionFilter(this);
+		}
+
+		/// <summary>
+		/// Returns the <c>Type</c> object which corresponds to the <c>MessageType</c> that 
+		/// is specified in the SubscriptionFilter.
+		/// </summary>
+		/// <remarks>
+		/// This method returns <c>null</c> for non-built in types (any type not
+		/// defined in SLNetTypes.dll or part of the Skyline.DataMiner.Net.Messages
+		/// namespace)
+		/// </remarks>
+		/// <returns></returns>
+		public Type ToTypeObject()
+		{
+			return ToType(MessageType);
+		}
+
+		/// <summary>
+		/// Gets the name of the type (e.g. AlarmEventMessage) that needs to be used in
+		/// a subscription filter if one wants to subscribe to the given type of messages.
+		/// </summary>
+		/// <param name="type">Type to get the name of.</param>
+		/// <returns></returns>
+		public static string ToTypeName(Type type)
+		{
+			return String.Empty;
+		}
+
+		/// <summary>
+		/// Converts a given type name (e.g. AlarmEventMessage) into the associated
+		/// Type object.
+		/// </summary>
+		/// <param name="messageType">Message type.</param>
+		/// <remarks>
+		/// This method returns <c>null</c> for non-built in types (any type not
+		/// defined in SLNetTypes.dll or part of the Skyline.DataMiner.Net.Messages
+		/// namespace)
+		/// </remarks>
+		/// <returns></returns>
+		public static Type ToType(string messageType)
+		{
+			// remove unwanted spaces (would cause GetType to fail)
+			return null;
+		}
+
+		/// <summary>
+		/// Base namespace for messages. If message types are within this namespace,
+		/// they can be qualified with their type name only, without the need
+		/// to specify the full type namespace.
+		/// </summary>
+		public const string BaseNameSpace = "Skyline.DataMiner.Net.Messages.";
+
+		/// <summary>
+		/// Message type of which all messages must derive.
```

Comment:
This seems a bit odd. Did you mean this?

```suggestion
		/// Message type from which all messages must derive.
```